modules/op-configuring-pipelines-as-code-logging.adoc

```diff
@@ -0,0 +1,124 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-pipelines-as-code.adoc
+
+:_content-type: PROCEDURE
+[id="configuring-pipelines-as-code-logging_{context}"]
+= Configuring {pac} logging 
+
+You can configure {pac} logging with the `pac-config-logging` config map located in the same namespace as the controllers. The default namespace is `openshift-pipelines`.
+
+.Procedure
+
+. Display the `pac-config-logging` config map:
++
+[source,terminal]
+----
+$ oc get configmap pac-config-logging -n openshift-pipelines
+----
++
+.Example output
+[source,terminal]
+----
+NAME                 DATA   AGE
+pac-config-logging   4      9m44s
+----
+
+. Retrieve the content of the `pac-config-logging` config map:
++
+[source,terminal]
+----
+$ oc get configmap pac-config-logging -n openshift-pipelines -o yaml
+----
++
+.Example output
+[source,terminal]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: pipelines-as-code
+  name: pac-config-logging
+  namespace: openshift-pipelines
+data:
+  loglevel.pac-watcher: info <1>
+  loglevel.pipelines-as-code-webhook: info <2>
+  loglevel.pipelinesascode: info <3>
+  zap-logger-config: |
+    {
+      "level": "info",
+      "development": false,
+      "sampling": {
+        "initial": 100,
+        "thereafter": 100
+      },
+      "outputPaths": ["stdout"],
+      "errorOutputPaths": ["stderr"],
+      "encoding": "json",
+      "encoderConfig": {
+        "timeKey": "ts",
+        "levelKey": "level",
+        "nameKey": "logger",
+        "callerKey": "caller",
+        "messageKey": "msg",
+        "stacktraceKey": "stacktrace",
+        "lineEnding": "",
+        "levelEncoder": "",
+        "timeEncoder": "iso8601",
+        "durationEncoder": "",
+        "callerEncoder": ""
+      }
+    }
+----
+<1> The log level for the `pipelines-as-code-watcher` component
+<2> The log level for the `pipelines-as-code-webhook` component
+<3> The log level for the `pipelines-as-code-controller` component
+
+. Change the log level to any supported value:
++
+[source,terminal]
+----
+$ oc patch configmap pac-config-logging -n openshift-pipelines --type json -p '[{"op": "replace", "path": "/data/loglevel.<component>", "value":"<value>"}]'
```

Comment:
@piyush-garg @savitaashture does this work ? don't we require users to update the tektonconfig object instead ?

Reply:
Yes because we don't update cm data using tektonconfig this is my understanding

Reply:
ok then 👼🏼

Reply:
@piyush-garg is that also true for Pipelines 1.12? We got informed before that the user will be able to modify this configmap directly in 1.12, that is why this issue exists for 1.12.

Reply:
i checked this, the value is getting reverted back in the configmap of the installerset, but not on the cluster. That seems like some other bug. But his will be reverted in case of upgrade atleast. So this approach might not work

Reply:
Hello @vdemeester What does this mean for the issue? Will it be accessible for the future versions or is this approach not planned? Because in that case, I would close this PR and close the issue as "won't do" or something similar.

Reply:
I think we need to document this (the behavior @piyush-garg highlighted) as a known issues.
@piyush-garg @savitaashture do we have an issue for it ?

Reply:
@vdemeester @piyush-garg @savitaashture question: can the log levels, and/or other settings in this configmap, be configured using TektonConfig? If they can. this is the preferred option anyway. But it they can not be configured by TektonConfig while configuration in the configmap gets overwritten, then it should be a known issue - and a corresponding bug Jira should be opened so Dev can provide the proper way of configuring logging.

Reply:
I think they cannot be configured through `TektonConfig` today, hence creating an issue and documenting that as a known issue or something.

Reply:
if it is only about log level of three components, maybe we can add them as part of spec.